### ISSUE TITLE
prometheus-cpp doesn't build

### ISSUE THREAD

**Issue by rsertelon** [jupp0r/prometheus-cpp#107]
Hi,
I'm trying to build 0.4 and get this error:

```
[ 43%] Building CXX object lib/CMakeFiles/prometheus-cpp.dir/summary.cc.o
In file included from /hab/cache/src/prometheus-cpp-0.4/lib/summary.cc:1:0:
/hab/cache/src/prometheus-cpp-0.4/lib/../include/prometheus/summary.h:49:61: error: field 'quantiles_' has incomplete type 'const std::reference_wrapper<const std::vector<prometheus::detail::CKMSQuantiles::Quantile> >'
   const std::reference_wrapper<const std::vector<Quantile>> quantiles_;
                                                             ^~~~~~~~~~
In file included from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/bits/move.h:54:0,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/bits/stl_pair.h:59,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/utility:70,
                 from /hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/array:38,
                 from /hab/cache/src/prometheus-cpp-0.4/lib/../include/prometheus/summary.h:3,
                 from /hab/cache/src/prometheus-cpp-0.4/lib/summary.cc:1:
/hab/pkgs/core/gcc/7.3.0/20180413214248/include/c++/7.3.0/type_traits:2125:11: note: declaration of 'class std::reference_wrapper<const std::vector<prometheus::detail::CKMSQuantiles::Quantile> >'
     class reference_wrapper;
           ^~~~~~~~~~~~~~~~~
make[2]: *** [lib/CMakeFiles/prometheus-cpp.dir/build.make:303: lib/CMakeFiles/prometheus-cpp.dir/summary.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:155: lib/CMakeFiles/prometheus-cpp.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

This is using GCC 7.3.0, cmake 3.10.2, make 4.2.1, glibc 2.27

**Comment by jupp0r** [comment]
Looks like we might be missing an `#include <functional>` in `summary.h`. Can you check if that solves the problem?

**Comment by jupp0r** [comment]
Fixed + released v0.4.1. Thanks for reporting this, I'll add GCC 7 builds to CI as a follow up to prevent this from happening in the future.

**Comment by rsertelon** [comment]
That is awesome! thanks a lot :)
